src/prof/src/http/mod.rs

```diff
@@ -199,7 +199,22 @@ mod disabled {
     ) -> impl IntoResponse {
         let merge_threads = threads.as_deref() == Some("merge");
         match action.as_ref() {
-            "time_fg" => Ok(time_prof(merge_threads, build_info, time_secs, hz).await),
+            "time_fg" => {
+                let time_secs = time_secs.ok_or_else(|| {
+                    (
+                        StatusCode::INTERNAL_SERVER_ERROR,
```

Comment:
Why we use `INTERNAL_SERVER_ERROR` and not `BAD_REQUEST` here?

Reply:
because I mindlessly copied what we were doing elsewhere in the function.

Good point though, I'll change it